managed/services/telemetry/telemetry.go

```diff
@@ -160,7 +160,47 @@ func (s *Service) DistributionMethod() serverpb.DistributionMethod {
 	return s.sDistributionMethod
 }
 
-func (s *Service) prepareReport(ctx context.Context) *reporter.ReportRequest {
+func (s *Service) processSendCh(ctx context.Context) {
+	var inflightReports []*pmmv1.ServerMetric
+	var sendCtx context.Context
+	var cancel context.CancelFunc
+
+	for {
+		select {
+		case report, ok := <-s.sendCh:
+			if ok {
+				inflightReports = append(inflightReports, report)
```

Comment:
So `inflightReports` can grow ethereally if we have some problem and some report always can't be sent?

Reply:
it will grow if it will be not possible to send, even after years of not sending it should now consume more than few megs of ram

Reply:
after successful send, it will be cleaned up

Reply:
Could you point to the logic where we clean up it? I can't find in the code where we clean up this slice.

Reply:
oh, I see, we should make sure that we won't have a race condition because of access to the same array from a multiple places.

Reply:
May be it's not for now, but I suggest to drop old telemetry data if queue size is exceeded.

Reply:
>Could you point to the logic where we clean up it? I can't find in the code where we clean up this slice.

L188

>we should make sure that we won't have a race condition because of access to the same array from a multiple places.

it is being copied before it is used, to avoid race

Reply:
but at line 188 we were setting nil value. that's what I meant. I see that you fixed it.